### PR TITLE
Update docs for 3.7.1

### PR DESCRIPTION
**Pull request by soarez**
This change should not to be merged before a 3.7.1 RC is promoted.

### CODE REVIEW

File: 37/generated/connect_metrics.html [hunk 1]
```diff
@@ -1,5 +1,5 @@
-[2024-02-23 00:02:00,837] INFO Metrics scheduler closed (org.apache.kafka.common.metrics.Metrics:694)
-[2024-02-23 00:02:00,838] INFO Metrics reporters closed (org.apache.kafka.common.metrics.Metrics:704)
+[2024-06-11 13:47:24,740] INFO Metrics scheduler closed (org.apache.kafka.common.metrics.Metrics:694)
+[2024-06-11 13:47:24,741] INFO Metrics reporters closed (org.apache.kafka.common.metrics.Metrics:704)
```
Comment (mimaison) [inline 37/generated/connect_metrics.html]: These lines at the top should not be there.
It looks like this has been fixed in trunk as I don't get them when I run `./gradlew siteDocsTar`, but I get them as well in 3.7.

Comment (soarez) [reply 37/generated/connect_metrics.html]: Thanks for pointing this out. I think this was fixed in apache/kafka#15473, so I'll backport that into the 3.7 branch.

File: 37/generated/admin_client_config.html [hunk 2]
```diff
@@ -284,7 +284,7 @@ <h4><a id="ssl.enabled.protocols"></a><a id="adminclientconfigs_ssl.enabled.prot
 <p>The list of protocols enabled for SSL connections. The default is 'TLSv1.2,TLSv1.3' when running with Java 11 or newer, 'TLSv1.2' otherwise. With the default value for Java 11, clients and servers will prefer TLSv1.3 if both support it and fallback to TLSv1.2 otherwise (assuming both support at least TLSv1.2). This default should be fine for most cases. Also see the config documentation for `ssl.protocol`.</p>
 <table><tbody>
 <tr><th>Type:</th><td>list</td></tr>
-<tr><th>Default:</th><td>TLSv1.2,TLSv1.3</td></tr>
```
Comment (mimaison) [inline 37/generated/admin_client_config.html]: I'm not sure we want these TLS config changes. It looks like the docs were generated with Java 8.

Comment (soarez) [reply 37/generated/admin_client_config.html]: Thanks for poiting this out. I'm not sure how this happened. I had another look to make the script is [using Java 17](https://github.com/apache/kafka/pull/16287/files#diff-241bcfc06fc20c009b77012a614513cc634006b8557f4d9cdcdc8fb8d44178e6R135).  Regenerated site docs, javadoc and rebased.

**Review by mimaison — APPROVED** [review]
LGTM

**Pull request merged**